src/Native/System.Private.CoreLib.Native/pal_dynamicload.cpp

```diff
@@ -0,0 +1,23 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include <dlfcn.h>
+
+extern "C" void* CoreLibNative_LoadLibrary(const char* filename)
+{
+    return dlopen(filename, RTLD_LAZY);
```

Comment:
Might need to prepend lib + append .so depending on the platform conventions here, but I don't have a Linux/OSX machine to test with so...